jax/numpy/lax_numpy.py

```diff
@@ -4574,6 +4575,21 @@ def _operator_round(number, ndigits=None):
 setattr(DeviceArray, "nbytes", property(_nbytes))
 
 
+# Experimental support for NumPy's module dispatch with NEP-37.
+# Currently requires https://github.com/seberg/numpy-dispatch
+_JAX_ARRAY_TYPES = (DeviceArray, core.Tracer)
```

Comment:
I'm not quite sure it this list should include `UnshapedArray` or one of JAX's other abstract array types.

Their presence in the set of types checked by `jax.numpy.ndarray` suggests yes, but on the other hand I don't think I've ever seen them in user supplied functions transformed with JAX?

Reply:
I like limiting the set for now. If it becomes clear later that `UnshapedArray` or others are necessary, they are easy enough to add.